src/main.ts

```diff
@@ -63,7 +63,7 @@ export async function run(argv = process.argv.slice(2), options?: Interfaces.Loa
   // find & run command
   const cmd = config.findCommand(id)
   if (!cmd) {
-    const topic = config.findTopic(id)
+    const topic = config.flexibleTaxonomy ? null : config.findTopic(id)
     if (topic) return config.runCommand('help', [id])
```

Comment:
Why's this necessary? Shouldn't the `help` command be able to be run on a topic?

For example:
![image](https://user-images.githubusercontent.com/1084688/157500716-9d16f52c-a28d-4d96-b65c-bc14e00ac33a.png)

Reply:
This code block is irrelevant in the case of `sf help env` since `help` is a valid command and this line of code is only hit if the command isn't found on the line above. 

The reason I added this was so that whenever someones executes a topic with flexible taxonomy enabled, oclif runs the `command_incomplete` hook instead of running help. Customers are still able to run `help env` and `env --help` as expected

![Screen Shot 2022-03-10 at 7 50 34 AM](https://user-images.githubusercontent.com/10244328/157687978-816c5d07-d86c-40d9-b698-da6933df6e47.png)